components/com_kunena/site/controller/widget/announcement/display.php

```diff
@@ -45,8 +45,9 @@ protected function before()
 			return false;
 		}
 
-		$view = JRequest::getWord ( 'view', 'default' );
-		$layout = JRequest::getWord ( 'layout', 'default' );
+		$jinput = JFactory::getApplication()->input;
+		$view = $jinput->getWord('view', 'default');
+		$layout = $jinput->getWord('layout', 'default');
```

Comment:
So:

    $view = $this->input->getWord('view', 'default');
    $layout = $this->input->getWord('layout', 'default');